mlir/include/mlir/Dialect/Mesh/Transforms/ReshardingSpmdizationDoc

```diff
@@ -0,0 +1,621 @@
+Reshadring Spmdization Examples
```

Comment:
I am not sure that this should be here. It is not a user facing doc, but it may be important to illustrate what is going on in resharding and what are the future challenges to make a communication efficient implementation.